src/utils/index.ts

```diff
@@ -103,6 +103,19 @@ const Utils = {
         }
         return inExtent
     },
+    // Returns true if the tile at xyz in the bounding box bb where bb is
+    // specified as [minEasting, minNorthing, maxEasting, maxNorthing]
+    // Checks to see if any corners of it fit inside
+    isInExtentEN: function (xyz: XYZ, bb, proj): boolean {
+        if (!bb) return true
+        const [minE, minN, maxE, maxN] = bb
+        const res = proj.res[xyz.z]
+        const origin = proj.tileMapResource.origin
+        const toEasting = x => x * proj.trueTileResolution * res + origin[0]
+        const toNorthing = y => -y * proj.trueTileResolution * res + origin[1]

```

Comment:
Nice! But could you use `proj.tileXYZ2NwSe()` (from projection.ts) instead -- I don't trust that the above formulas would work in every use case whereas I'm much more confident the `tileXYZ2NwSe` function with `asBounds=true` will since it uses proj4.

Otherwise it looks great!

Reply:
Thanks @tariqksoliman! I just changed it to use `tileXYZ2NwSe`, but while testing with my application, it is requesting slightly too many tiles because some tiles that perfectly line up with the edges of the bounds are off by a very small number (something like 0.0000000004).

Would it be fine if I add a small margin to the boundingBox in this function? Or is this something I should add to my application code?

Reply:
I like the idea and adding the margins to the extent functions directly makes the most sense I think.